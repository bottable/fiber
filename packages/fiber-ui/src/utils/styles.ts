import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
