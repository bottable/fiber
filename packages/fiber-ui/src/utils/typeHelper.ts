export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
