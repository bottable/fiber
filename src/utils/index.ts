export * from './typeHelper'
export * from './mergeProps'
