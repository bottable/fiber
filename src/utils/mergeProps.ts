import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
