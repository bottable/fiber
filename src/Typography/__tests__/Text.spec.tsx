import { Text } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Text', () => {
  it('renders correctly', () => {
    const TextGroup = render(
      <>
        <Text> This is a text </Text>
        <Text strong> This is a strong text </Text>
        <Text highlight> This is a highlighted text </Text>
        <Text code> const fiber = "hello world" </Text>
        <Text size={4}> This is a text of size 4. </Text>
        <Text size={5} highlight code>
          This is a code of size 5, highlighted.
        </Text>
      </>
    )

    expect(TextGroup).toMatchSnapshot()
  })
})
