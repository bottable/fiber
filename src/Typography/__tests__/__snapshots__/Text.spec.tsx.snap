// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
Array [
  <span
    className="sc-AxirZ hqkDwY"
    size={4}
  >
     This is a text 
  </span>,
  <span
    className="sc-AxirZ hpLNBn"
    size={4}
  >
     This is a strong text 
  </span>,
  <span
    className="sc-AxirZ bidIDo"
    size={4}
  >
     This is a highlighted text 
  </span>,
  <span
    className="sc-AxirZ hDfFCg"
    size={4}
  >
     const fiber = "hello world" 
  </span>,
  <span
    className="sc-AxirZ hqkDwY"
    size={4}
  >
     This is a text of size 4. 
  </span>,
  <span
    className="sc-AxirZ isPnJF"
    size={5}
  >
    This is a code of size 5, highlighted.
  </span>,
]
`;
